drivers/video/video_mcux_smartdma.c

```diff
@@ -267,7 +267,12 @@ static int nxp_video_sdma_get_format(const struct device *dev, struct video_form
 	if ((fmt->pixelformat != fmts[0].pixelformat) ||
 	    (fmt->width != fmts[0].width_min) ||
 	    (fmt->height != fmts[0].height_min)) {
-		return -ENOTSUP;
+		ret = video_set_format(config->sensor_dev, fmt);
+
+		if (ret < 0) {
+			LOG_ERR("Sensor device does not support RGB565");
+			return ret;
+		}
```

Comment:
That indeed fixes the bug encountered by the capture sample.

The fix for this is ultimately in the `capture` sample IMHO, but for the sake of a workaround that stll makes sense.

Another compromise in the meantime is to configure the default format in `_init()` and make this call always return the default supported format without asking the source?

Reply:
If you wanted to make sure that `fmt = RGB565 320x240`, you would also need to use `fmts[0]` instead of `fmt`: as this is:

```c
if (/* format is not what is expected */) {
    /* use the incorrect fmt */
}
```

So that would need a `tmp = { /* correct fmt */ }` if wanting to go that path (but +1 for simply moving that in `init()`).

Reply:
Ah I think I see your way and I prefer it, so at unit we can set format to fmts[0], I think that’s better because this is the default format  for SDMA and if anything deviates it should be flagged rather than coercing an error into the right thing. Will push this change shortly.

Reply:
@ngphibang  makes sense shall do this now.

Reply:
@ngphibang  I just did this and there is now just one commit I didn't move the work to _init as if it is initialized before the sensor then it won't work cc @josuah